packages/editor/src/components/post-sync-status/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { ToggleControl, PanelRow } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+export default function PostSyncStatus() {
+	const { editPost } = useDispatch( editorStore );
+	const { meta, postType } = useSelect( ( select ) => {
+		const { getEditedPostAttribute } = select( editorStore );
+		return {
+			meta: getEditedPostAttribute( 'meta' ),
+			postType: getEditedPostAttribute( 'type' ),
+		};
+	}, [] );
+	if ( postType !== 'wp_block' ) {
+		return null;
+	}
+	const onUpdateSync = ( syncStatus ) =>
+		editPost( {
+			meta: { ...meta, wp_block: { sync_status: syncStatus } },
+		} );
+	const syncStatus = meta?.wp_block?.sync_status;
+	return (
+		<PanelRow className="edit-post-sync-status">
+			<span>{ __( 'Sync status' ) }</span>
```

Comment:
Have just reduced it to `Syncing` for the user facing label for now. If we get approval for this approach I think I will change `syncStatus` to `syncType` everywhere as `syncStatus` is a good fit - it sounds more like whether the pattern `has` been synced rather than `should` it be synced.